google/cloud/iam/integration_tests/iam_integration_test.cc

```diff
@@ -514,6 +514,17 @@ TEST_F(IamIntegrationTest, GetServiceAccountProtoFailure) {
   EXPECT_THAT(log_lines, Contains(HasSubstr("GetServiceAccount")));
 }
 
+template <typename T>
+T StatusRetryLoop(std::function<T(void)> const& operation) {
```

Comment:
What you want here is something like:

```cc
template <typename Functor>
invoke_result_t<Functor> StatusRetryLoop(Functor&& operation) {
  using ReturType = invoke_result_t<Functor>;
  ReturnType result;
  for (auto delay : ...) {
    result = operator();
 ...
  }
  return result;
}
```

Now, `std::invoke_result_t<>` is a C++17 feature, but you can find a friendly replacement at:

https://github.com/googleapis/google-cloud-cpp/blob/7e8bc7c22790a8cbbde211e618be08702afdab5c/google/cloud/internal/invoke_result.h#L171